core/src/main/java/org/elasticsearch/action/search/SearchPhaseExecutionException.java

```diff
@@ -85,8 +85,15 @@ private static Throwable deduplicateCause(Throwable cause, ShardSearchFailure[]
     @Override
     public RestStatus status() {
```

Comment:
Are there any tests for this method?
It looks like the code only expects a SearchPhaseExecutionException only when there are shard failure(s). The bug was that an exception occurred on the coordinating node performing reduce, which resulted in this code assuming all shards failed (503 status instead of the expected root cause status).